pkgs/applications/science/misc/openmodelica/mkderivation/src-main.nix

```diff
@@ -0,0 +1,7 @@
+{
+  url = "https://github.com/OpenModelica/OpenModelica/";
+  rev = "08fd3f9144235f209a4ed7602bfadb32b1823628";
+  sha256 = "0clgqk9ilnr43iyl5sdzwfzqpnw9amfy1npdgkpgm1wfnsvz6xrw";
+  fetchSubmodules = true;
+}
+# Update with: nix run -f ./nixpkgs/default.nix nix-prefetch-git -c nix-prefetch-git 'https://github.com/OpenModelica/OpenModelica/' 'v1.17.0' --fetch-submodules
```

Comment:
Please move this into the derivation and use fetchFromGitHub.

Reply:
Sorry, that is impossible for two reasons:
* `fetchFromGitHub` does not fetch submodules for OpenModelica. I have tried that earlier.
* We use `src-main.nix` in two places -- in `mkderivation/default.nix` and in `omlibrary/update-src-libs.sh`. So better that to be in a separate file.